src/averageAges.js

```diff
@@ -37,7 +42,16 @@ function calculateMenAverageAge(people, century) {
  * @return {number}
  */
 function calculateWomenAverageAge(people, withChildren) {
-  // write code here
+  const women = withChildren
+    ? people.filter((p) => p.sex === 'f'
+      && people.find((p1) => p1.mother === p.name))
+    : people.filter((p) => p.sex === 'f');
+
+  let sum = 0;
+
+  women.forEach((w) => sum += w.died - w.born);
```

Comment:
It is better to do it with help of `reduce` array method
See [this](https://mate-academy.github.io/style-guides/javascript.html#111-dont-use-iterators-prefer-javascripts-higher-order-functions-instead-of-loops-likefor-inorfor-of) documentation